### ISSUE TITLE
Does betterproto support nested list of list ?

### ISSUE THREAD

**Issue by luochen1990** [danielgtaylor/python-betterproto#290]
As we know, Protobuf's syntax doesn't support [generic type](https://stackoverflow.com/questions/12305318/how-to-handle-the-generic-type-object-with-protocol-buffers-in-the-proto-file/34291587) or [list of list](https://stackoverflow.com/questions/36715078/google-protobuffer-how-to-define-list-of-lists-in-proto-file).

But it seems that there is no limitation in the betterproto-generated code:

```
@dataclass
class Msg(betterproto.Message):
    msg: str = betterproto.message_field(1)
    info: List["Info"] = betterproto.message_field(2)
```

So I'm wondering, is it possible to hand-write the betterproto-generated code and define nested list ?

just like:

```
@dataclass
class Msg(betterproto.Message):
    msg: str = betterproto.message_field(1)
    info: List[List["Info"]] = betterproto.message_field(2)
```

Will this class which contains `List[List[xxx]]` be serialized and deserialized normally?
In other words, whether the limitation of protobuf is only on the syntax layer or there is a deeper reason?

**Comment by Gobot1234** [comment]
There is no special casing for this and afaik this doesn't work.
